File: app/src/main/java/closer/vlllage/com/closer/ui/FitsSystemWindowsConstraintLayout.java
package closer.vlllage.com.closer.ui;

import android.content.Context;
import android.support.constraint.ConstraintLayout;
import android.util.AttributeSet;
import android.view.WindowInsets;

public class FitsSystemWindowsConstraintLayout extends ConstraintLayout {
    public FitsSystemWindowsConstraintLayout(Context context) {
        super(context);
        init();
    }

    public FitsSystemWindowsConstraintLayout(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public FitsSystemWindowsConstraintLayout(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    private void init() {
        setFitsSystemWindows(true);
    }

    @Override
    public WindowInsets onApplyWindowInsets(WindowInsets insets) {
        super.onApplyWindowInsets(insets);
        return insets;
    }
}

File: app/src/main/java/closer/vlllage/com/closer/ui/FitsSystemWindowsConstraintLayout.kt
package closer.vlllage.com.closer.ui

import android.content.Context
import androidx.constraintlayout.widget.ConstraintLayout
import android.util.AttributeSet
import android.view.WindowInsets

class FitsSystemWindowsConstraintLayout : ConstraintLayout {
    constructor(context: Context) : super(context) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        init()
    }

    private fun init() {
        fitsSystemWindows = true
    }

    override fun onApplyWindowInsets(insets: WindowInsets): WindowInsets {
        super.onApplyWindowInsets(insets)
        return insets
    }
}
